### PR TITLE
2.4 bridge name 1804018

### PR DESCRIPTION
**Pull request by jameinel**
## Description of change

This is rebasing https://github.com/juju/juju/pull/9482 onto 2.4 and fixing up the test suite.

It changes how we create bridge names for bridges put on top of devices that have vlans. We essentially never put a '.' into a bridge name, so that it bridges to the proper device, rather than bridging to the underlying device and adding vlan on top of it.

## QA steps

See the bug report.

I did not test handling of VLAN'd bridges directly, as I don't have an appropriate setup to test that packets actually get routed correctly.

I did test that I was still able to create a machine and put containers on it, and it would create a bridge for those containers.

## Documentation changes

None.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1804018

### CODE REVIEW

**Comment by jameinel** [comment]
[1804018 autogenerated bridge name and VLAN](https://trello.com/c/gl7kocHh/808-1804018-autogenerated-bridge-name-and-vlan)

**Comment by jameinel** [comment]
!!build!!

**Comment by jameinel** [comment]
(windows build failed with:
Cloning into '/home/ubuntu/go/src/github.com/rogpeppe/godeps'...
fatal: unable to access 'https://github.com/rogpeppe/godeps/': Could not resolve host: github.com
)

File: network/containerizer/bridgepolicy.go [hunk 1]
```diff
@@ -180,14 +180,16 @@ var skippedDeviceNames = set.NewStrings(
 // 2. Add b- to device name, if it doesn't fit in 15 characters then:
 // 3a. For devices starting in 'en' remove 'en' and add 'b-'
 // 3b. For all other devices 'b-' + 6-char hash of name + '-' + last 6 chars of name
+// 4. If using the device name directly always replace '.' with '-', to make sure that bridges from VLANs won't break
 func BridgeNameForDevice(device string) string {
+	device = strings.Replace(device, ".", "-", -1)
```
Comment (manadart) [inline network/containerizer/bridgepolicy.go]: Do we need to do this both at the top and for each of the cases below?

**Comment by jameinel** [comment]
$$merge$$

**Pull request merged**